pkg/volume/csi/expander.go

```diff
@@ -62,5 +69,28 @@ func (c *csiPlugin) NodeExpand(spec *volume.Spec, devicePath, deviceMountPath st
 	if err != nil {
 		return false, fmt.Errorf("Expander.NodeExpand failed to check if node supports expansion : %v", err)
 	}
-	return false, nil
+
+	if !nodeExpandSet {
+		return false, fmt.Errorf("Expander.NodeExpand found CSI plugin %s to not support node expansion", c.GetPluginName())
+	}
+
+	// Check whether "STAGE_UNSTAGE_VOLUME" is set
+	stageUnstageSet, err := csiClient.NodeSupportsStageUnstage(ctx)
+	if err != nil {
+		return false, fmt.Errorf("Expander.NodeExpand failed to check if plugins supports stage_unstage %v", err)
+	}
+
+	// if plugin does not support STAGE_UNSTAGE but CSI volume path is staged
+	// it must mean this was placeholder staging performed by k8s and not CSI staging
+	// in which case we should return from here so as volume can be node published
+	// before we can resize
+	if !stageUnstageSet && resizeOptions.CSIVolumePhase == volume.CSIVolumeStaged {
```

Comment:
If the volume plugin does support stage, but the phase is node publish, then do we want to still call resize?

Reply:
This function could use some unit tests

Reply:
ideally - if plugin indeed supports stage then call should be made in staging phase. But say when volume is already node staged and node published (and hence in publish phase) then resizing can indeed be done in publish phase too. From CSI specs:

```
If plugin has STAGE_UNSTAGE_VOLUME node capability then:

    NodeExpandVolume MUST be called after successful NodeStageVolume.
    NodeExpandVolume MAY be called before or after NodePublishVolume.
```

Reply:
I have added some unit tests.